### PR TITLE
Support bundle for mgmt cluster

### PR DESCRIPTION
**Pull request by danbudris**
*Description of changes:*
Previously, we assumed all clusters were self-managed when generating a support bundle.

Now, we check the `isSelfManaged()` method of the spec to determine if the we should collect and analyze management resources (CAPI crds, cluster crd, bundle, etc) when generating a support bundle for a cluster.

the method `WithManagementCluster(isSelfManaged bool)` will determine if a bundle uses the mgmt cluster collectors/analyzers or not.

### Testing
generated default support bundles w/out spec input
generated support bundles w/spec input
generated support bundles for workload and management clusters

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/548#" title="Author self-approved">danbudris</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jiayiwang7 — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/548#" title="Author self-approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/548#pullrequestreview-795853082" title="Approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
